### PR TITLE
perf(behavior_velocity_planner): remove pose.orientation from occlusion_spot

### PR DESCRIPTION
**Pull request by veqcc**
Signed-off-by: Ryuta Kambe [ryuta.kambe@tier4.jp](mailto:ryuta.kambe@tier4.jp)

## Description
cc: @taikitanaka3 @soblin
In occlusion_spot module in behavior_velocity_planner node, `pose` of collision information are calculated, though those `pose` are used only for extracting `pose.position` information. `pose.orientation` are wastefully calculated and they have a lot of computational cost. Thus I deleted that calculation and replaced `calcPose` function to `calcPosition`.

## Related links
https://github.com/autowarefoundation/autoware.universe/issues/2097

### CODE REVIEW

**Comment by veqcc** [comment]
The red line is the original latency of occlusion_spot, and the blue one is the latency when this commit is adapted.
image

**Comment by taikitanaka3** [comment]
@veqcc 
is virtual wall refactoring necessary with this PR?I think  this PR should include part of this PR 
https://github.com/autowarefoundation/autoware.universe/pull/2506

File: planning/behavior_velocity_planner/include/scene_module/occlusion_spot/occlusion_spot_utils.hpp [hunk 1]
```diff
@@ -183,13 +183,15 @@ struct PossibleCollisionInfo
...
   std::string detection_type = "";
   Polygons2d detection_area_polygons;
   std::vector<lanelet::BasicPolygon2d> close_partition;
   std::vector<geometry_msgs::msg::Point> parked_vehicle_point;
   std::vector<PossibleCollisionInfo> possible_collisions;
   std::vector<geometry_msgs::msg::Point> occlusion_points;
+  std::vector<geometry_msgs::msg::Pose> debug_poses;
   void resetData()
   {
     close_partition.clear();
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/include/scene_module/occlusion_spot/occlusion_spot_utils.hpp:197]: This change is also necessary to remove previous marker
```suggestion
    debug_poses.clear();
    close_partition.clear();
```

And I tested with this change.
image

**Review by soblin — APPROVED** [review]

**Comment by soblin** [comment]
@veqcc 
Please run
```
git rebase HEAD~4 --signoff
git push --force-with-lease origin remove_pose_orientation_from_occlusion_spot
```
to pass the check

**Pull request merged**